src/model/HeatPumpPlantLoopEIRCooling.cpp

```diff
@@ -0,0 +1,491 @@
+      return getAutosizedValue("Design Size Load Side Volume Flow Rate", "m3/s");
+    }
+
+    boost::optional<double> HeatPumpPlantLoopEIRCooling_Impl::autosizedReferenceSourceSideFlowRate() const {
+      return getAutosizedValue("Design Size Source Side Volume Flow Rate", "m3/s");
+    }
+
+    boost::optional<double> HeatPumpPlantLoopEIRCooling_Impl::autosizedReferenceCapacity() const {
+      return getAutosizedValue("Design Size Nominal Capacity", "W");
+    }
+
+    void HeatPumpPlantLoopEIRCooling_Impl::autosize() {
+      autosizeReferenceLoadSideFlowRate();
+      autosizeReferenceSourceSideFlowRate();
+      autosizeReferenceCapacity();
+    }
+
+    void HeatPumpPlantLoopEIRCooling_Impl::applySizingValues() {
+      boost::optional<double> val;
+      val = autosizedReferenceLoadSideFlowRate();
+      if (val) {
+        setReferenceLoadSideFlowRate(val.get());
+      }
+
+      val = autosizedReferenceSourceSideFlowRate();
+      if (val) {
+        setReferenceSourceSideFlowRate(val.get());
+      }
+
+      val = autosizedReferenceCapacity();
+      if (val) {
+        setReferenceCapacity(val.get());
+      }
+    }
+
+  }  // namespace detail
+
+  HeatPumpPlantLoopEIRCooling::HeatPumpPlantLoopEIRCooling(const Model& model, const CurveBiquadratic& capacityModifierFunctionofTemperatureCurve,
+                                                           const CurveBiquadratic& electricInputtoOutputRatioModifierFunctionofTemperatureCurve,
+                                                           const CurveQuadratic& electricInputtoOutputRatioModifierFunctionofPartLoadRatioCurve)
+    : WaterToWaterComponent(HeatPumpPlantLoopEIRCooling::iddObjectType(), model) {
+    OS_ASSERT(getImpl<detail::HeatPumpPlantLoopEIRCooling_Impl>());
+
+    autosizeReferenceLoadSideFlowRate();
+    autosizeReferenceSourceSideFlowRate();
+    autosizeReferenceCapacity();
+
+    bool ok = setCapacityModifierFunctionofTemperatureCurve(capacityModifierFunctionofTemperatureCurve);
+    OS_ASSERT(ok);
+
+    ok = setElectricInputtoOutputRatioModifierFunctionofTemperatureCurve(electricInputtoOutputRatioModifierFunctionofTemperatureCurve);
+    OS_ASSERT(ok);
+
+    ok = setElectricInputtoOutputRatioModifierFunctionofPartLoadRatioCurve(electricInputtoOutputRatioModifierFunctionofPartLoadRatioCurve);
+    OS_ASSERT(ok);
+
+    setCondenserType("WaterSource");
+    setReferenceCoefficientofPerformance(7.5);
+    setSizingFactor(1.0);
+  }
+
+  HeatPumpPlantLoopEIRCooling::HeatPumpPlantLoopEIRCooling(const Model& model)
+    : WaterToWaterComponent(HeatPumpPlantLoopEIRCooling::iddObjectType(), model) {
+    OS_ASSERT(getImpl<detail::HeatPumpPlantLoopEIRCooling_Impl>());
+
+    autosizeReferenceLoadSideFlowRate();
+    autosizeReferenceSourceSideFlowRate();
+    autosizeReferenceCapacity();
+
+    CurveBiquadratic capacityModifierFunctionofTemperatureCurve(model);  // PlantLoopHeatPump_EIR_WaterSource.idf
```

Comment:
Lol. So all three curves are flatlines at 1. (I know this is the E+ example and not your doing)